js/src/Selector.ts

```diff
@@ -96,7 +97,7 @@ export abstract class BaseXSelector extends BaseSelector {
         if(this.model.get("scale")) {
             const that = this;
             return this.create_child_view(this.model.get("scale")).then(function(view) {
-                that.scale = view;
+                that.scale = view as WidgetView as LinearScale;
```

Comment:
You can have double `as`?
Are you qualifying a `WidgetView` as a `LinearScale` here?

Reply:
Yes, casting `view` to `LinearScale` is not accepted because the objects are too different if I understand correctly, so you need to cast `view` to `WidgetView` first, then `WidgetView` to `LinearScale`.

Reply:
You can specify to `create_child_view` all the right types, something like:

```
return this.create_child_view<LinearScaleModel>(this.model.get("scale")).then(function(view: LinearScale) {
```

Reply:
Good to know, thanks!